master/pkg/schemas/expconf/experiment_config_test.go

```diff
@@ -10,6 +10,159 @@ import (
+    environment:
+      environment_variables: {}
+      force_pull_image: false
+      image:
+        cpu: determinedai/environments:py-3.7-pytorch-1.7-lightning-1.2-tf-2.4-cpu-da845fc
+        gpu: determinedai/environments:cuda-11.0-pytorch-1.7-lightning-1.2-tf-2.4-gpu-da845fc
+      pod_spec: null
+      ports:
+        qwer: 1234
+        asdf: 5678
+      registry_auth:
+        username: usr
+        password: pwd
+        auth: ath
+        email: eml
+        serveraddress: srvaddr
+        identitytoken: idtkn
+        registrytoken: rgtkn
+      add_capabilities:
+        - CAP_CHOWN
+      drop_capabilities:
+        - CAP_KILL
+    hyperparameters:
+      global_batch_size:
+        type: const
+        val: 32
+      list_hparam:
+        - 10
+        - type: const
+          val: asdf
+        - type: int
+          minval: 1
+          maxval: 2
+      dict_hparam:
+        double_hparam:
+          type: double
+          minval: 1
+          maxval: 10
+        log_hparam:
+          type: log
+          minval: 1
+          maxval: 10
+          base: 1
+      categorical_hparam:
+        type: categorical
+        vals: [1, 2, 3, 4]
+      nested_hparam:
+        hp1: 32
+    labels: []
+    max_restarts: 5
+    min_validation_period:
+      batches: 0
+    name: pytorch-noop
+    optimizations:
+      aggregation_frequency: 1
+      auto_tune_tensor_fusion: false
+      average_aggregated_gradients: true
+      average_training_metrics: false
+      gradient_compression: false
+      grad_updates_size_file: "/tmp/hi I am a size file"
+      mixed_precision: O0
+      tensor_fusion_cycle_time: 5
+      tensor_fusion_threshold: 64
+    perform_initial_validation: false
+    profiling:
+      enabled: true
+      begin_on_batch: 0
+      end_after_batch: 1
+    records_per_epoch: 0
+    reproducibility:
+      experiment_seed: 1606239866
+    resources:
+      agent_label: 'big_al'
+      devices:
+        - host_path: "/dev/infiniband"
+          container_path: "/dev/infiniband"
+          mode: "rmw"
+      slots_per_trial: 15
+      weight: 1000
+      max_slots: 900
+      priority: 55
+      resource_pool: 'asdf'
+      native_parallel: false
+    scheduling_unit: 100
+    searcher:
+      max_length:
+        batches: 1000
+      metric: loss
+      name: single
+      smaller_is_better: true
+      source_checkpoint_uuid: null
+      source_trial_id: null
+    security:
+      kerberos:
+        config_file: xyz
+    tensorboard_storage:
+      type: shared_fs
+      host_path: /tmp
+      container_path: /asdf
+      checkpoint_path: /qwer
+      tensorboard_path: /zxcv
+            `),
+			Expected: ExperimentConfig{},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.Name, func(t *testing.T) {
+			jByts, err := schemas.JSONFromYaml(tc.Bytes)
+			assert.NilError(t, err)
+			config, err := ParseAnyExperimentConfigYAML(jByts)
+			assert.NilError(t, err)
+			err = schemas.IsComplete(config)
```

Comment:
Can you just add nested hp to the 'full experiment' test in `schemas/test_cases/v0/experiment.yaml`?

All this does is unmarshal and check IsComplete, but it only does it for one language.  Putting the test in `schemas/test_cases` will ensure that when the python sdk supports ExperimentConfig objects that we have all the same test coverage on both languages.